app/services/jwt_token_validate.rb

```diff
@@ -0,0 +1,66 @@
+require 'jwt'
+require 'rest-client'
+
+class JwtTokenValidate
+  attr_reader :decoded_token
+
+  def initialize(jwt_token, url=nil, realm=nil)
+    @jwt_token = jwt_token
+    @url = url
+    @realm = realm
+  end
+
+  def valid_jwt?
+    #TODO check for JWKS and aud check
+    valid_signature? && type_bearer?
+  rescue JWT::ImmatureSignature
+    Rails.logger.error "JWT SSO: JWT is for the future."
+    false
+  rescue JWT::ExpiredSignature
+    Rails.logger.error "JWT SSO: Signature has expired."
+    false
+  rescue JWT::DecodeError
+    Rails.logger.error "JWT SSO: Failed to decode JWT."
+    false
+  end
+
+  def generate_jwks_uri
+    response = RestClient::Request.execute(
+      :url => "https://ssoo1.usersys.redhat.com/auth/realms/hammer-cli/protocol/openid-connect/certs",
```

Comment:
I have added this to the settings, this URL is different for each user, so it may differ. To obtain the URL user can read the description of the field as seen [here](https://github.com/theforeman/foreman/pull/6549/files#diff-6afdff346c7e417ea7f74f437487706aR30)